src/rbindlist.c

```diff
@@ -282,7 +282,7 @@ SEXP rbindlist(SEXP l, SEXP usenamesArg, SEXP fillArg, SEXP idcolArg)
     for (int i=0; i<LENGTH(l); ++i) {
       SEXP li = VECTOR_ELT(l, i);
       if (!length(li)) continue;
-      int w = usenames ? colMap[i*ncol + j] : j;  // colMap tells us which item to fetch for each of the final result columns, so we can stack column-by-column
+      int w = usenames ? colMap[i*ncol + j] : (j<length(li) ? j : -1);  // colMap tells us which item to fetch for each of the final result columns, so we can stack column-by-column // check if j exceeds length for fill=TRUE and usenames=FALSE #5444
```

Comment:
nit: why the second comment marker (`//`)?

Reply:
I looked at that as well. If comments are unrelated then better to have second // so it's clear for whoever reads that it's completely new comment. Otherwise comments could be merged into single one.

Reply:
Comments are unrelated, why I went for the second //

Reply:
makes sense... i think elsewhere that's where we'd see

```c
foo(); // comment
       // another comment
```

side note -- for me personally, that's why I usually prefer anchoring the comment to the previous line rather than end-of-line, makes it easier: continuations get a hanging indent, new comments restart indentation:

```c
// this is a comment that
//   spills onto line 2
// this is an unrelated comment
```